library/alloc/src/gc.rs

```diff
@@ -75,6 +74,104 @@ static FINALIZERS_REGISTERED: AtomicU64 = AtomicU64::new(0);
 #[cfg(profile_gc)]
 static FINALIZERS_COMPLETED: AtomicU64 = AtomicU64::new(0);
 
+#[derive(Debug)]
+pub struct GcAllocator;
```

Comment:
Should we document this struct as something like "internal only: calling any of its functions yourself is UB"?

Reply:
Good point. Now that we've moved bits around, it's clear that those "internal-only" methods shouldn't actually belong to that allocator. I'll move them out so that only the ones that should be re-exported via `std::gc::GcAllocator` remain.